File: AlgoWorks/src/com/solid/algolearning/javacode/algorithms/hackerRankChallenges/Factorial.java
package com.solid.algolearning.javacode.algorithms.hackerRankChallenges;

public class Factorial {

    public static void main(String[] args) {
        System.out.println(findFactorialIterative(5));
        System.out.println(findFactorialRecursive(5));
    }

    public static int findFactorialIterative(int number){
        int answer = 1;

        if (number == 2){
            answer = 2;
        }

        while (number > 0){
            answer *= number;
            number--;
        }

        return answer;
    }

    public static int findFactorialRecursive(int number){
        if (number == 0 || number == 1){
            return 1;
        }

        return number * findFactorialRecursive(number - 1);
    }
}

File: AlgoWorks/src/com/solid/algolearning/kotlincode/algorithms/factorial.kt
package com.solid.algolearning.kotlincode.algorithms

fun main() {
        println(findFactorialIterative(5))
        println(findFactorialRecursive(5))
}

fun findFactorialIterative(_number: Int) : Int{
    var answer = 1
    var number = _number

    if (number == 2){
        answer = 2
    }

    while (number > 0){
        answer *= number
        number--
    }

    return answer
}

fun findFactorialRecursive(_number: Int): Int{
    var number = _number

    if (number == 0 || number == 1){
        return 1
    }

    return number * findFactorialRecursive(number - 1)
}